backend/FwLite/MiniLcm/Filtering/EntryFilter.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Linq.Expressions;
+using Gridify;
+using Gridify.Syntax;
+using MiniLcm.Models;
+
+namespace MiniLcm.Filtering;
+
+public class EntryFilter
+{
+    public static GridifyMapper<Entry> NewMapper()
+    {
+        var mapper = new GridifyMapper<Entry>(false);
+        mapper.AddMap(nameof(Entry.ComplexFormTypes));
+        mapper.AddMap($"{nameof(Entry.Senses)}.{nameof(Sense.SemanticDomains)}");
+        mapper.AddMap($"{nameof(Entry.Senses)}.{nameof(Sense.ExampleSentences)}");
+        mapper.AddMap($"{nameof(Entry.Senses)}.{nameof(Sense.PartOfSpeechId)}");
+        mapper.AddMap($"{nameof(Entry.Senses)}.{nameof(Sense.Gloss)}",
+            (entry, key) => entry.Senses.Select(s => s.Gloss[key]));
+        mapper.AddMap($"{nameof(Entry.Senses)}.{nameof(Sense.Definition)}",
+            (entry, key) => entry.Senses.Select(s => s.Definition[key]));
+        mapper.AddMap(nameof(Entry.Senses));
+        mapper.AddMap(nameof(Entry.LexemeForm), (entry, key) => entry.LexemeForm[key]);
+        mapper.AddMap(nameof(Entry.CitationForm), (entry, key) => entry.CitationForm[key]);
+        mapper.AddMap(nameof(Entry.Note), (entry, key) => entry.Note[key]);
+        return mapper;
+    }
+
+    //used by the database for json columns which are lists, we want to treat null as an empty list
+    public static object ConvertNullToEmptyList<T>(string value)
+    {
+        if (value is "null" or "") return new List<T>();
+        throw new Exception($"Invalid value {value} for {typeof(T).Name}");
+    }
```

Comment:
I don't get this.
Here's an example usage: `Mapper.AddMap(nameof(Entry.ComplexFormTypes), EntryFilter.ConvertNullToEmptyList<ComplexFormType>);`

Why doesn't the exception get hit when `Entry.ComplexFormTypes` is a non-empty list?

Reply:
Yeah, this is a bit of a hack. This is for converting the filter value. Eg if the filter string is `ComplexFormTypes=null` then this function will be called with `value=null`. So if you tried to filter `ComplexFormTypes=banana` then this exception would be thrown, because what does that even mean? This is where we could say `ComplexFormTypes=[]` and add that to the if statement. Maybe we should just do that?

Reply:
OK. I get it.
Yeah, I think `[]` is a pretty reasonable value to handle.

I went ahead and added it.